man/mergelist.Rd

```diff
@@ -0,0 +1,202 @@
+\name{mergelist}
+\alias{mergelist}
+\alias{setmergelist}
+\title{Merge multiple data.tables}
+\description{
+  Faster merge of multiple \code{data.table}s.
+}
+\usage{
+  mergelist(l, on, cols=NULL,
+    how=c("left", "inner", "full", "right", "semi", "anti", "cross"),
+    mult, join.many=getOption("datatable.join.many"))
+  setmergelist(l, on, cols=NULL,
+    how=c("left", "inner", "full", "right", "semi", "anti", "cross"),
+    mult, join.many=getOption("datatable.join.many"))
+}
+\arguments{
+  \item{l}{ \code{list} of \code{data.table}s to merge. }
+  \item{on}{ \code{character} vector of column names to merge on; when missing, the \code{\link{key}} of the \emph{join-to} table is used (see Details). }
+  \item{cols}{ Optional \code{list} of \code{character} column names corresponding to tables in \code{l}, used to subset columns during merges. \code{NULL} means all columns, all tables; \code{NULL} entries in a list means all columns for the corresponding table. }
+  \item{how}{ Character string, controls how to merge tables. Allowed values are \code{"left"} (default), \code{"inner"}, \code{"full"}, \code{"right"}, \code{"semi"}, \code{"anti"}, and \code{"cross"}. See Details. }
+  \item{mult}{ Character string, controls how to proceed when multiple rows in the \emph{join-to} table match to the row in the \emph{join-from} table. Allowed values are \code{"error"}, \code{"all"}, \code{"first"}, \code{"last"}. The default value depends on \code{how}; see Details. See Examples for how to detect duplicated matches. When using \code{"all"}, we recommend specifying \code{join.many=FALSE} as a precaution to prevent unintended explosion of rows. }
+  \item{join.many}{ \code{logical}, defaulting to \code{getOption("datatable.join.many")}, which is \code{TRUE} by default; when \code{FALSE} and \code{mult="all"}, an error is thrown when any \emph{many-to-many} matches are detected between pairs of tables. This is essentially a stricter version of the \code{allow.cartesian} option in \code{\link{[.data.table}}. Note that the option \code{"datatable.join.many"} also controls the behavior of joins in \code{[.data.table}. }
+}
+\details{
+  Note: these functions should be considered experimental. Users are encouraged to provide feedback in our issue tracker.
```

Comment:
should we leave this note when merging to master?

Reply:
Yes, why not for 2-3 major releases